### PR TITLE
refactor: refact crd based harborconfiguration

### PR DESCRIPTION
**Pull request by chlins**
Fixes: https://github.com/goharbor/harbor-operator/issues/724, https://github.com/goharbor/harbor-operator/issues/723

Signed-off-by: chlins <chlins.zhang@gmail.com>

### CODE REVIEW

File: apis/goharbor.io/v1beta1/harbor_configuration.go [hunk 1]
```diff
@@ -1,10 +1,10 @@
 package v1beta1
 
 import (
-	"encoding/json"
-
+	goyaml "gopkg.in/yaml.v2"
```
Comment (cndoit18) [inline apis/goharbor.io/v1beta1/harbor_configuration.go:4]: Why not use v3?

Comment (chlins) [reply apis/goharbor.io/v1beta1/harbor_configuration.go:4]: https://github.com/go-yaml/yaml

File: apis/goharbor.io/v1beta1/harbor_configuration.go [hunk 2]
```diff
@@ -32,233 +32,228 @@ type HarborConfiguration struct {
...
 	// +kubebuilder:validation:Optional
-	StoragePerProject int `json:"storage_per_project,omitempty"`
+	StoragePerProject int `json:"storagePerProject,omitempty" yaml:"storage_per_project,omitempty"`
 }
 
 // ToJSON converts configuration spec to json payload.
-func (h HarborConfigurationSpec) ToJSON() ([]byte, error) {
-	jsonData, err := json.Marshal(h)
+func (h HarborConfigurationModel) ToJSON() ([]byte, error) {
+	data, err := goyaml.Marshal(h)
```
Comment (cndoit18) [inline apis/goharbor.io/v1beta1/harbor_configuration.go:95]: May `apiextensionsv1.JSON` also be parsed correctly?

**Review by heww — APPROVED** [review]

**Review by bitsf — APPROVED** [review]

File: apis/goharbor.io/v1beta1/harbor_configuration.go (hunk 2 above)
Comment (chlins) [reply apis/goharbor.io/v1beta1/harbor_configuration.go:95]: Yes, tested.

**Pull request merged**